[reland][CI] switch torchbench to a pinned version

desertfire opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #96782

Summary: This is reland of https://github.com/pytorch/pytorch/pull/96553

cc @soumith @voznesenskym @penguinwu @anijain2305 @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx

desertfire committed:
[reland][CI] switch torchbench to a pinned version

Summary: This is reland of https://github.com/pytorch/pytorch/pull/96553

[ghstack-poisoned]

weiwangmeta commented on benchmarks/dynamo/check_hf_bert_perf_csv.py:
```diff
@@ -17,7 +17,8 @@ def check_hf_bert_perf_csv(filename):
         # Reduce from 1.165 to 1.160, see https://github.com/pytorch/pytorch/issues/96530
-        if speedup < 1.160:
+        # Reduce from 1.160 to 1.145 after a transformer version upgrade, see https://github.com/pytorch/benchmark/pull/1406
+        if speedup < 1.145:
```
During re-land, let's change it to 1.140, e.g. this action generated 1.1405 https://github.com/pytorch/pytorch/actions/runs/4420433748/jobs/7750698385

huydhn commented on .ci/pytorch/test.sh:
```diff
@@ -320,12 +320,7 @@ test_single_dynamo_benchmark() {
-        --actual "$TEST_REPORTS_DIR/${name}_$suite.csv" \
-        --expected "benchmarks/dynamo/ci_expected_accuracy/${name}_${suite}${shard_id}.csv"
-    fi
+    # will add graph_break check back after collecting new graph break csv
```
Will did a forward fix here https://github.com/pytorch/pytorch/pull/96780, so the conflict would need to be resolved

huydhn reviewed (approved):
Pre-stamp approval

desertfire committed:
Update on "[reland][CI] switch torchbench to a pinned version"


Summary: This is reland of https://github.com/pytorch/pytorch/pull/96553

cc soumith voznesenskym penguinwu anijain2305 EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx

[ghstack-poisoned]

desertfire commented:
@pytorchbot merge -g

desertfire commented:
@pytorchbot merge -f "PR only affects torchbench tests and they have passed"
